tests/compiler/test_basics.py

```diff
@@ -212,3 +218,116 @@ def test_value_set_twice(snippetcompiler):
+    v2_template_path: str = os.path.join(modules_v2_dir, "minimalv2module")
+    libs_dir: str = os.path.join(str(tmpdir), "libs")
+
+    # main module, used as proxy to load test module's model and plugins
+    main_module: str = "main"
+    # module under test
+    test_module: str = "test_module"
+
+    def test_module_plugin_contents(value: int) -> str:
+        """
+        Returns the contents of the plugin file for the test module.
+        """
+        return f"""
+from inmanta.plugins import plugin
+
+
+VALUE: str = {value}
+
+@plugin
+def get_value() -> "int":
+    return VALUE
+        """.strip()
+
+    # The model for the test module, regardless of value
+    test_module_model: str = f"value = {test_module}::get_value()"
+
+    # install main module (make it v1 so there are no restrictions on what it can depend on)
+    v1_module_from_template(
+        v1_template_path,
+        os.path.join(libs_dir, f"{main_module}"),
+        new_name=main_module,
+        new_content_init_cf=f"""
+import {test_module}
+
+# test variable import
+test_module_value = {test_module}::value
+# test plugin call
+test_module_value = {test_module}::get_value()
+# test intermodule Python imports
+test_module_value = {main_module}::get_test_module_value()
+        """.strip(),
+        new_content_init_py=f"""
+from inmanta.plugins import plugin
+from {const.PLUGINS_PACKAGE}.{test_module} import VALUE
+
+
+@plugin
+def get_test_module_value() -> "int":
+    return VALUE
+        """.strip(),
+    )
+
+    def verify_compile(expected_value: int) -> None:
+        """
+        Verify compilation by importing main module and checking its variable's value.
+        """
+        snippetcompiler_clean.setup_for_snippet(
+            f"""
+import {main_module}
+
+# make sure imported variable has expected value (these statements will produce compiler error if it doesn't)
+value = {main_module}::test_module_value
+value = {expected_value}
+            """.strip(),
+            add_to_module_path=[libs_dir],
+            # set autostd=False because no v2 std exists at the time of writing
+            autostd=False,
+            install_project=False,
+        )
+        compiler.do_compile()
+
+    # install test module as v1 and verify compile
+    v1_module_from_template(
+        v1_template_path,
+        os.path.join(libs_dir, test_module),
+        new_name=test_module,
+        new_content_init_cf=test_module_model,
+        new_content_init_py=test_module_plugin_contents(1),
+    )
+    verify_compile(1)
+
+    # install module as v2 (on top of v1) and verify compile
+    v2_module_path: str = os.path.join(str(tmpdir), test_module)
+    module_from_template(
+        v2_template_path,
+        v2_module_path,
+        new_name=test_module,
+        new_content_init_cf=test_module_model,
+        new_content_init_py=test_module_plugin_contents(2),
+        install=True,
+        editable=True,
```

Comment:
I would add a comment that this is editable 

```suggestion
        editable=True,  # ! this is editable for the next test step
```